src/compositions/post/form-post.tsx

```diff
@@ -1,35 +1,34 @@
-import { createPost } from '@/app/actions/post';
-import { auth } from '@/app/api/auth/[...nextauth]/auth';
 import { Avatar, AvatarSize } from '@ost-cas-fee-adv-23-24/design-system-pixelpioneers';
 import WritePost from '../write-post/write-post';
-import { PostFormTypeVariant } from './types';
+import { ActionTypeVariant, PostFormTypeVariant } from './types';
 import { getUser } from '@/app/actions/user';
 import { User } from '@/src/models/user.model';
+import { Session } from 'next-auth';
 
-export default async function NewPost() {
-    const session = await auth();
-
+export default async function NewPost({ session }: { session: Session | null }) {
```

Comment:
call this FormPost or rename the file to new-post.. or a totally different name like PostForm?